### DISCUSSION TITLE
Custom server with multiple next.js apps hangs on startup after upgrading from 9.3.3 to 9.3.4

### DISCUSSION THREAD

**Discussion by kidrecursive** [vercel/next.js#12084] [Help]
# Bug report

## Describe the bug

A custom server that creates/prepares two or more NextJS apps hangs after upgrading from 9.3.3 to 9.3.4.

## To Reproduce

Example: https://github.com/kidrecursive/nextjs-multiple-apps 

Clone the repository, install the dependencies, and run `npm start`. The `prepare` promises for the two next apps never resolve. Roll back next with `npm install next@9.3.3`, remove the `.next` build folders from both apps, and restart; the server starts and serves both apps.

## Expected behavior

The server starts and serves both next.js apps

## System information

- OS: OSX 10.15.4
- Version of Next.js: 9.3.4
- Version of Node.js: 12.16.1

**Comment by AdamusPiotr** [comment]
We have the same problem, i debugged it a little bit and it seems to stop at enabling hot reload, probably it want's to register hooks on the same port's as the first application

**Comment by kidrecursive** [comment]
I looked into this further and updated the original issue. One thing I left out in recreating was the step of deleting the `.next` build folders inside of the apps after changing the next.js version.

https://github.com/zeit/next.js/releases/tag/v9.3.4 is the tag that appears to have the changes in it that prevent the previous behavior from working, I'll take a look here next.

**Reply by pelhage** [reply]
@kidrecursive any updates? Curious to explore non-conventional nextjs custom server set ups. Multiple apps seems to be a step in the right direction for me

I wish the build dir could be externally hosted, i.e. on a CDN and dynamically set at runtime

**Reply by kidrecursive** [reply]
Heya, in the most recent version things work as expected (in production), and when in dev mode, you are able to only view one of the apps at a time. I've been working around this so far and haven't revisited.

**Comment by kavyantic** [comment]
I am also stucked in the same issue and could not find the solution yet. My NEXTjs version is 12.2.5, nextApp.prepare() never resolves in my AWS instance and it works absolutely fine as expected in my local windows machine and can't rollback to any older version as already utilizing latest features.

````

const mobileNextApp = next({ dev, dir: mobileDir ectory});
const nextHandleMobile = mobileNextApp.getRequestHandler();

//other devices frontend init
const nextApp = next({ dev, dir: basicNextDirctory });
const nextHandle = nextApp.getRequestHandler();

nextApp.prepare().then(() => {
  console.log("next js for other devices prepared");
  mobileNextApp.prepare().then(() => {
    console.log("nextjs for mobile prepared successfully");
    console.log('\x1b[32m', 'successfully started all UI services');

    expressServer.all("*", (req, res) => {
      if (isMobile(req.headers["user-agent"])) {
        console.log("mobile request");
        return nextHandleMobile(req, res);
      }
      return nextHandle(req, res);
    });
  });
});`
```
